frontend/src/components/review/CommentThread.tsx

```diff
@@ -0,0 +1,119 @@
+import React, { useState } from "react";
+import { useMutation } from "@apollo/client";
+import ExistingComment from "./ExistingComment";
+import WIPComment from "./WIPComment";
+import { CommentResponse } from "../../APIClients/queries/CommentQueries";
+import {
+  UPDATE_COMMENT_BY_ID,
+  UpdateCommentResponse,
+} from "../../APIClients/mutations/CommentMutations";
+
+export type CommentThreadProps = {
+  comment: CommentResponse;
+  comments: CommentResponse[];
+  setComments: (comments: CommentResponse[]) => void;
+  threadReplies: CommentResponse[];
+  updateCommentsAsResolved: (index: number) => void;
+  updateThreadHeadMap: (cmts: CommentResponse[]) => void;
+  userName: string;
+};
+
+const CommentThread = ({
+  comment,
```

Comment:
since you're getting `comments` prop, you could further simplify these props by accepting `commentsIdx` to retrieve `comment = comments[commentsIdx]`. You could also figure out `threadReplies` from `comments` + `commentsIdx`

Reply:
Then, in `CommentsPanel` you just need to generate CommentThread's for any `true` comments in the threadheadmap